metrics/ci_integrations/test/cli/logger/logger_test.dart

```diff
@@ -1,122 +1,126 @@
-import 'dart:convert';
 import 'dart:io';
 
 import 'package:ci_integration/cli/logger/logger.dart';
+import 'package:intl/intl.dart';
+import 'package:mockito/mockito.dart';
 import 'package:test/test.dart';
 
-// ignore_for_file: avoid_redundant_argument_values
+import '../test_util/stub/io_sink_stub.dart';
 
 void main() {
   group("Logger", () {
     final errorSink = IOSinkStub();
     final messageSink = IOSinkStub();
+    final unimplementedSink = IOSinkStub(
+      writelnCallback: (_) => throw UnimplementedError(),
+    );
+    const message = 'message';
+    const error = 'error';
 
     test(
-      "creates Logger with default stderr if no error sink is given",
+      ".logError() throws an Exception if the Logger is not configured",
       () {
-        final logger = Logger(messageSink: messageSink);
-        final errorSink = logger.errorSink;
-
-        expect(errorSink, equals(stderr));
+        expect(() => Logger.logError(error), throwsException);
       },
     );
 
     test(
-      "creates Logger with default stdout if no message sink is given",
+      ".logMessage() throws an Exception if the Logger is not configured",
       () {
-        final logger = Logger(errorSink: errorSink);
-        final messageSink = logger.messageSink;
+        expect(() => Logger.logMessage(message), throwsException);
+      },
+    );
 
-        expect(messageSink, equals(stdout));
+    test(
+      ".logInfo() throws an Exception if the Logger is not configured",
+      () {
+        expect(() => Logger.logInfo(message), throwsException);
       },
     );
 
-    test(".printError() prints the given error to the error sink", () {
-      final unimplementedMessageSink = IOSinkStub(
-        writelnCallback: (_) => throw UnimplementedError(),
-      );
-      final logger = Logger(
+    test(".setup() configure the logger with the given error sink", () {
+      final errorSink = IOSinkMock();
+      Logger.setup(errorSink: errorSink);
+
+      Logger.logError(error);
+
+      verify(errorSink.writeln(any)).called(1);
+    });
+
+    test(".setup() configure the logger with the given message sink", () {
+      final messageSink = IOSinkMock();
+      Logger.setup(messageSink: messageSink);
+
+      Logger.logMessage(message);
+
+      verify(messageSink.writeln(any)).called(1);
+    });
+
+    test(".logError() prints the given error to the error sink", () {
+      Logger.setup(
         errorSink: errorSink,
-        messageSink: unimplementedMessageSink,
+        messageSink: unimplementedSink,
       );
-      const expected = 'error';
 
       expect(
-        () => logger.printError(expected),
-        prints(equalsIgnoringWhitespace(expected)),
+        () => Logger.logError(error),
+        prints(equalsIgnoringWhitespace(error)),
```

Comment:
Would it be more reliable to check that this method adds only a newline symbol at the end of the given value and prints the given value without any modifications? 
```suggestion
        prints(equals('$error\n')),
```